### PR TITLE
docs: some work on moving over the reference docs

### PR DESCRIPTION
**Pull request by orta**
I've started porting over the reference docs from the shiki-twoslash playground!

One tricky deprecation blocker so far, which is the auto-complete query doesn't work because it can't grab the character _after_ the last one

```
console.e
//       ^|
```
=>
```
Error: [shikiji-twoslash] Cannot find token at L0:9
    at locateTextToken (file:///home/orta/dev/typescript/twoslashes/node_modules/.pnpm/shikiji-twoslash@0.10.0-beta.0/node_modules/shikiji-twoslash/dist/core.mjs:869:19)
    at Object.code (file:///home/orta/dev/typescript/twoslashes/node_modules/.pnpm/shikiji-twoslash@0.10.0-beta.0/node_modules/shikiji-twoslash/dist/core.mjs:897:25)
````

Which is kinda true, (there is no token!) but the auto-complete is only relevant after the text

### CODE REVIEW

**Comment by antfu** [comment]
Oh I wasn't aware that use cases, added a test to guard it: 70f6d1c1f96e1abca5a0ab8ddae10052436d5b63

It seems the output of two slash is expected; I will handle it on the Shikiji side.

**Comment by antfu** [comment]
I actually fixed it already on the last Shikiji release :P, you may need to `pnpm i` to update them :)

File: docs/refs/notations.md [hunk 1]
```diff
@@ -1,3 +1,130 @@
...
+// ---cut---
+// @filename: b.ts
+import { helloWorld } from './a'
+
+console.log(helloWorld)
+```
+
+Would only show the last two lines, but to TypeScript it was a program with 2 files and all of the IDE information is hooked up correctly across the files. This is why `// @filename: [file]` is specifically the only Twoslash command which _is not_ removed, because if it's not relevant it can be `---cut---` away.
+
+### `---cut-after---`
```
Comment (antfu) [inline docs/refs/notations.md:59]: I also supported `---cut-start---` and `---cut-end---` pairs, which I think would be useful :) Feel free to add them, or I can do it later :)

File: docs/refs/notations.md [hunk 2]
```diff
@@ -1,3 +1,130 @@
...
+
+The sibling to `--cut---` which trims anything after the sigil:
```
Comment (antfu) [inline docs/refs/notations.md:61]:
```suggestion
The sibling to `---cut---` which trims anything after the sigil:
```

**Review by antfu — APPROVED** [review]

File: docs/refs/notations.md (hunk 1 above)
Comment (orta) [reply docs/refs/notations.md:59]: Yeah, open to `---cut-start---` being an "alias" to `---cut---`

**Comment by orta** [comment]
Nice

**Pull request merged**